app/api/dao/user.py

```diff
@@ -173,10 +282,16 @@ def confirm_registration(token):
 
     @staticmethod
     def authenticate(username_or_email, password):
-        """
+        """Authenticates a user.
+
+        Authenticates a user given the latter has a valid email and provides the appropriate password.
         The user can login with two options:
-        -> username + password
-        -> email + password
+            -> username + password
+            -> email + password
+
+        Args:
+            username_or_email : username or email of the user being authenticated.
+            password: List containing the current and new password and their corresponding values.
```

Comment:
you are missing the return part here. This function either returns a user or None, please explain that following the doc string guide.